### PR TITLE
Set required Hydra property when relation is not nullable.

### PR DESCRIPTION
**Pull request by raoulclais**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #1567
| License       | MIT
| Doc PR        |

Implement #1567 by checking if the relation is required.

### CODE REVIEW

File: tests/Bridge/Doctrine/Orm/Metadata/Property/DoctrineOrmPropertyMetadataFactoryTest.php [hunk 1]
```diff
@@ -59,10 +60,11 @@ public function testCreateNoClassMetadata()
         $this->assertEquals($doctrineOrmPropertyMetadataFactory->create(Dummy::class, 'id', []), $propertyMetadata);
     }
 
-    public function testCreateIsIdentifier()
+    public function testCreateIsIdentifierNotRequired()
     {
         $propertyMetadata = new PropertyMetadata();
         $propertyMetadata = $propertyMetadata->withIdentifier(true);
+        $propertyMetadata = $propertyMetadata->withRequired(false);
```
Comment (soyuka) [inline tests/Bridge/Doctrine/Orm/Metadata/Property/DoctrineOrmPropertyMetadataFactoryTest.php]: Bu definition an identifier is required but I think I get what you did :p.

**Comment by dunglas** [comment]
Shouldn't required properties be marked as `@NotNull` or `@NotBlank` with the validator? If we go this way, we must also mark properties with `nullable=false` as required... and enforce it somewhere because it's not the case yet (it's the responsibility of the validator, not of the ORM in a typical Symfony app).

**Comment by dunglas** [comment]
I'm for closing this one as "won't fix". Ping @api-platform/core-team.

**Comment by soyuka** [comment]
Also if you really need the feature, you can always add a custom MetadataPropertyFactory right?

**Pull request closed without merging**